nautobot/utilities/utils.py

```diff
@@ -576,3 +578,37 @@ def is_uuid(value):
     except (ValueError, TypeError, AttributeError):
         pass
     return False
+
+
+def pretty_print_query(query):
+    """
+    Given a `Q` object, display it in a more human-readable format.
+
+    :param query:
+        Q instance
+    """
+
+    def __our_str__(self):
+        """Improvement to default `Node.__str__` with a more human-readable style."""
+        template = "(NOT %s)" if self.negated else "(%s)"
+        children = []
+
+        # Iterate over children. They will be either a Q object (a Node subclass) or a 2-tuple.
+        for child in self.children:
+            # Trust that we can stringify the child if it is a Node instance.
+            if isinstance(child, Node):
+                children.append(str(child))
+            # If a 2-tuple, stringify to key=value
+            else:
+                key, value = child
+                children.append(f"{key}={value!r}")
+
+        return template % (f" {self.connector} ".join(children))
+
+    # Swap `Node.__str__` for our version and replace it when we're done.
+    try:
+        orig_str = query.__class__.__str__
+        query.__class__.__str__ = __our_str__
+        return str(query)
+    finally:
+        query.__class__.__str__ = orig_str
```

Comment:
Does this really need to monkey-patch `__str__`, or could it be as simple as:

```python
def pretty_print_query(query):
    def our_str(self):
        ...
        if isinstance(child, Node):
            children.append(our_str(child))
        ...
```

Reply:
I'll give it a whirl.